editor.planx.uk/src/@planx/components/FindProperty/Public/Map.tsx

```diff
@@ -0,0 +1,136 @@
+          longitude: geojson["EPSG:3857"].features[0]?.geometry?.coordinates[0],
+          latitude: geojson["EPSG:3857"].features[0]?.geometry?.coordinates[1],
+          x: geojson["EPSG:27700"].features[0]?.geometry?.coordinates[0],
+          y: geojson["EPSG:27700"].features[0]?.geometry?.coordinates[1],
+          title: siteDescription || "",
+          source: "proposed",
+        });
+      } else {
+        // triggered if a user "clears" their point on the map
+        setProposedAddress(undefined);
+      }
+    };
+
+    const map: any = document.getElementById("plot-new-address-map");
+    map?.addEventListener("geojsonChange", geojsonChangeHandler);
+
+    return function cleanup() {
+      map?.removeEventListener("geojsonChange", geojsonChangeHandler);
+    };
+  }, [setProposedAddress]);
+
+  return (
+    <Card
+      handleSubmit={() => {
+        //@ts-ignore
+        props.setAddress({ ...proposedAddress, title: siteDescription });
+      }}
+      isValid={Boolean(proposedAddress) && Boolean(siteDescription)}
+    >
+      <QuestionHeader
+        title={props.title || DEFAULT_NEW_ADDRESS_TITLE}
+        description={props.description || ""}
+      />
+      <MapContainer environment={environment} interactive>
+        <p style={visuallyHidden}>
+          An interactive map centred on the local authority district, showing
+          the Ordnance Survey basemap. Click to place a point representing your
+          proposed site location.
+        </p>
+        {/* @ts-ignore */}
+        <my-map
+          id="plot-new-address-map"
+          data-testid="map-web-component"
+          zoom={14}
+          drawMode
+          drawType="Point"
+          resetControlImage="trash"
+          showScale
+          showNorthArrow
+          osVectorTilesApiKey={process.env.REACT_APP_ORDNANCE_SURVEY_KEY}
+        />
+        <MapFooter>
+          <Typography variant="body2">
+            The coordinate location of your address point is:{" "}
+            <strong>
+              {proposedAddress?.x?.toFixed(5) || 0} Easting (X),{" "}
+              {proposedAddress?.y?.toFixed(5) || 0} Northing (Y)
+            </strong>
+          </Typography>
+          <Link
+            component="button"
+            onClick={() => props.setPage("os-address")}
+            disabled={false}
+          >
+            <Typography variant="body2">
+              I want to select an existing address
+            </Typography>
+          </Link>
+        </MapFooter>
+      </MapContainer>
+      <DescriptionInput data-testid="new-address-input">
+        <InputLabel label="Describe this site">
+          <Input
+            name="newAddress"
+            bordered
+            onChange={(e) => {
+              setSiteDescription(e.target.value);
+            }}
+            value={siteDescription}
+            id="newAddressInput"
+          />
+        </InputLabel>
```

Comment:
I imagine a nice v2.0 improvement here would be to additionally make X & Y _inputs_ - allowing you to enter values and the map marker would respond. But I've defaulted to non-inputs for this MVP (it's not a need described in the Trello ticket yet), and only map interaction, to simplify component interactions and back/"change" behavior for now

Reply:
I think of this as a nice v2 feature also...!